static/app/views/issueDetails/groupDetails.tsx

```diff
@@ -312,41 +341,51 @@ function useFetchGroupDetails({
     refetch: refetchEvent,
   } = useEventApiQuery(eventId, [eventUrl, {query: eventQuery}]);
 
+  const {
+    data: groupData,
+    isLoading: loadingGroup,
+    isError: isGroupError,
+    error: groupError,
+    refetch: refetchGroupCall,
+  } = useApiQuery<Group>(makeFetchGroupQueryKey({groupId, environments}), {
+    staleTime: 30000,
+    cacheTime: 30000,
+    enabled: isGlobalSelectionReady,
+  });
+
+  const {data: groupReleaseData, isLoading: groupReleaseLoading} =
+    useApiQuery<GroupRelease>([`/issues/${groupId}/first-last-release/`], {
+      staleTime: 60000,
+      cacheTime: 60000,
+      enabled: !!groupData,
+    });
+
   useEffect(() => {
-    if (eventData) {
-      setEvent(eventData);
+    if (!groupReleaseLoading && groupReleaseData) {
+      GroupStore.onPopulateReleases(groupId, groupReleaseData);
     }
-  }, [eventData]);
+  }, [groupReleaseData, groupReleaseLoading, groupId]);
 
-  const fetchGroupReleases = useCallback(async () => {
-    const releases = await api.requestPromise(
-      `/issues/${params.groupId}/first-last-release/`
-    );
-    GroupStore.onPopulateReleases(params.groupId, releases);
-  }, [api, params.groupId]);
+  const group = groupData ?? null;
 
-  const handleError = useCallback((e: RequestError) => {
-    Sentry.captureException(e);
-
-    setLoadingGroup(false);
-    setErrorType(getFetchDataRequestErrorType(e?.status));
-    setError(true);
-  }, []);
+  useSyncGroupStore(environments);
 
-  const fetchData = useCallback(async () => {
-    // Need to wait for global selection store to be ready before making request
-    if (!isGlobalSelectionReady) {
-      return;
+  useEffect(() => {
+    if (eventData) {
+      setEvent(eventData);
     }
+  }, [eventData]);
 
-    try {
-      const groupResponse = await api.requestPromise(`/issues/${params.groupId}/`, {
-        query: getGroupQuery({environments}),
-      });
-      fetchGroupReleases();
+  useEffect(() => {
+    if (!loadingGroup && group) {
+      GroupStore.loadInitialData([group]);
```

Comment:
I found a bug where sometimes this line would fire twice, once before onPopulateReleases and once after. I _believe_ this occurs when the group releases response is still cached, but the group details needs to be refetched.

To ensure that `loadInitialData` never gets called after `onPopulateReleases`, I'd try putting both of them in the same useEffect.